src/main/java/appeng/me/NetworkEventBus.java

```diff
@@ -114,10 +115,19 @@ MENetworkEvent postEvent( final Grid g, final MENetworkEvent e )
 						target.invoke( cache.getCache(), e );
 					}
 
-					for( final IGridNode obj : g.getMachines( subscriber.getKey() ) )
+					// events may create or remove grid nodes in rare cases
+					final IMachineSet machines = g.getMachines( subscriber.getKey() );
+					final List<IGridNode> work = new ArrayList<>( machines.size() );
+					machines.forEach( work::add );
+
+					for( final IGridNode obj : work )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)